packages/create-turbo/src/commands/create/index.ts

```diff
@@ -0,0 +1,231 @@
+
+  let project: Project = {} as Project;
+  try {
+    project = await getWorkspaceDetails({ root });
+  } catch (err) {
+    handleWorkspaceErrors(err);
+  }
+
+  let successfulTransforms = 0;
+  if (!skipTransforms) {
+    for (const transform of transforms) {
+      try {
+        const result = await transform({
+          example: {
+            repo: repoInfo,
+            name: exampleName,
+          },
+          project,
+          prompts: {
+            projectName,
+            root,
+            packageManager: selectedPackageManagerDetails,
+          },
+          opts,
+        });
+        if (result.result === "success") {
+          successfulTransforms += 1;
+        }
+      } catch (err) {
+        handleWorkspaceErrors(err);
+      }
+    }
+  }
+
+  if (successfulTransforms > 0) {
+    // create a second commit after running transforms
+    tryGitCommit("feat(create-turbo): apply transforms");
+  }
+
+  // if the user opted out of transforms, the package manager will be the same as the example
+  const projectPackageManager =
+    skipTransforms || !selectedPackageManagerDetails
+      ? {
+          name: project.packageManager,
+          version: availablePackageManagers[project.packageManager].version,
+        }
+      : selectedPackageManagerDetails;
+
+  info("Created a new Turborepo with the following:");
+  console.log();
+  if (project.workspaceData.workspaces.length > 0) {
+    const workspacesForDisplay = project.workspaceData.workspaces
+      .map((w) => ({
+        group: path.relative(root, w.paths.root).split(path.sep)?.[0] || "",
+        title: path.relative(root, w.paths.root),
+        description: w.description,
+      }))
+      .sort((a, b) => a.title.localeCompare(b.title));
+
+    let lastGroup: string | undefined;
+    workspacesForDisplay.forEach(({ group, title, description }, idx) => {
+      if (idx === 0 || group !== lastGroup) {
+        console.log(chalk.cyan(group));
+      }
+      console.log(
+        ` - ${chalk.bold(title)}${description ? `: ${description}` : ""}`
+      );
+      lastGroup = group;
+    });
+  } else {
+    console.log(chalk.cyan("apps"));
+    console.log(` - ${chalk.bold(projectName)}`);
+  }
+
+  console.log();
+  if (hasPackageJson && !skipInstall) {
+    if (
+      opts.skipTransforms &&
+      !availablePackageManagers[project.packageManager].available
+    ) {
+      warn(
+        `Unable to install dependencies - "${exampleName}" uses "${project.packageManager}" which could not be found.`
+      );
+      warn(
+        `Try running without "--skip-transforms" to convert "${exampleName}" to a package manager that is available on your system.`
+      );
+      console.log();
+    } else if (selectedPackageManagerDetails) {
+      console.log("Installing packages. This might take a couple of minutes.");
+      console.log();
+
+      const loader = turboLoader("Installing dependencies...").start();
+      await install({
```

Comment:
Does installing dependencies create a lockfile or is there already one? In either case, I think this is also a transform. (Especially if we every want to think about `turbo install` as a feature)

Reply:
Yea I did think about this, and It does produce a lock if one doesn't exist. However, I would argue that's not a transform, it's just a by-product of what you asked for via arguments. There's already another option to enable / disable this behavior (`skip-install`)